File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.py
class Solution:
    def deckRevealedIncreasing(self, deck: list[int]) -> list[int]:
        deck.sort()
        res = [0] * len(deck)  # 0 0 0 0 0 0 0
        queue = deque(range(len(deck)))  # 0 1 2 3 4 5 6

        for num in deck:
            i = queue.popleft()
            res[i] = num
            if queue:
                queue.append(queue.popleft())

        return res

File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.js
/**
 * @param {number[]} deck
 * @return {number[]}
 */
var deckRevealedIncreasing = function (deck) {
    deck.sort((a, b) => a - b)
    n = deck.length
    let result = new Array(n).fill(0)
    let queue = [] //stores the indices
    for (let i = 0; i < n; i++) {
        queue.push(i)
    }

    for (let num of deck) {
        //popleft 1st time
        let i = queue.shift()
        result[i] = num
        if (queue.length) {
            //popleft 2nd time : add to the back
            queue.push(queue.shift())
        }

    }

    return result

};